iocore/net/quic/QUICLossDetector.cc

```diff
@@ -390,12 +391,12 @@ QUICLossDetector::_detect_lost_packets(QUICPacketNumberSpace pn_space)
 
       if (unacked->in_flight) {
         lost_packets.insert({it->first, it->second.get()});
-      } else if (this->_loss_time[static_cast<int>(pn_space)] == 0) {
-        this->_loss_time[static_cast<int>(pn_space)] = unacked->time_sent + loss_delay;
-      } else {
-        this->_loss_time[static_cast<int>(pn_space)] =
-          std::min(this->_loss_time[static_cast<int>(pn_space)], unacked->time_sent + loss_delay);
       }
+    } else if (this->_loss_time[static_cast<int>(pn_space)] == 0) {
+      this->_loss_time[static_cast<int>(pn_space)] = unacked->time_sent + loss_delay;
```

Comment:
Bug

Reply:
Is this just a mistake, or does the spec have a bug?

Reply:
mistake

```
   DetectLostPackets(pn_space):
     loss_time[pn_space] = 0
     lost_packets = {}
     loss_delay = kTimeThreshold * max(latest_rtt, smoothed_rtt)

     // Packets sent before this time are deemed lost.
     lost_send_time = now() - loss_delay

     // Packets with packet numbers before this are deemed lost.
     lost_pn = largest_acked_packet[pn_space] - kPacketThreshold

     foreach unacked in sent_packets:
       if (unacked.packet_number > largest_acked_packet[pn_space]):
         continue

       // Mark packet as lost, or set time when it should be marked.
       if (unacked.time_sent <= lost_send_time ||
           unacked.packet_number <= lost_pn):
         sent_packets.remove(unacked.packet_number)
         if (unacked.in_flight):
           lost_packets.insert(unacked)
       else:
         if (loss_time[pn_space] == 0):
           loss_time[pn_space] = unacked.time_sent + loss_delay
         else:
           loss_time[pn_space] = min(loss_time[pn_space],
                                     unacked.time_sent + loss_delay)

     // Inform the congestion controller of lost packets and
     // let it decide whether to retransmit immediately.
     if (!lost_packets.empty()):
       OnPacketsLost(lost_packets)
```